bindings/gumjs/gumcmodule.c

```diff
@@ -8,6 +8,91 @@
 
 #include <gio/gio.h>
 
+struct _GumCModuleOps
+{
+  GumCModule * (*gum_cmodule_new) (const GumCModuleOps *, const gchar *,
+      GError **);
+  void (*gum_cmodule_free) (GumCModule *);
+  void (*gum_cmodule_add_symbol) (GumCModule *, const gchar *, gconstpointer);
+  gint (*gum_cmodule_link_pre) (GumCModule *, GString **);
+  gint (*gum_cmodule_link) (GumCModule *, gpointer, GString **);
+  void (*gum_cmodule_link_post) (GumCModule *);
+  void (*gum_cmodule_enumerate_symbols) (GumCModule *, GumFoundCSymbolFunc,
+      gpointer);
+  gpointer (*gum_cmodule_find_symbol_by_name) (GumCModule *, const gchar *);
+  void (*gum_cmodule_drop_metadata) (GumCModule *);
+  void (*gum_cmodule_add_define) (GumCModule *, const gchar *, const gchar *);
+};
```

Comment:
It would be good to define a GObject interface instead, and implement it in two different GObject-classes. For an example of an interface see `gum/gumbacktracer.h` and `gum/gumbacktracer.c`.

Reply:
Should GObject be the new API, or can it live purely inside gumcmodule.c as an implementation detail?

Reply:
I think keeping it inside is fine. Can be split into multiple files later if we end up adding more backends.